src/tests/unit/components/sidebar.spec.js

```diff
@@ -0,0 +1,63 @@
+import { shallowMount, createLocalVue } from "@vue/test-utils";
+import Sidebar from "../../../components/Sidebar.vue";
+import store from "../../../store";
+import VueRouter from "vue-router";
+
+const localVue = createLocalVue()
+localVue.use(VueRouter)
+const router = new VueRouter()
+
+describe("Sidebar", () => {
+  const wrapper = shallowMount(Sidebar, {
+    stubs: ["router-link"],
+    store,
+    router,
+  });
+
+  // get sidebarItems from getters
+  // find out how many .parent_box we have
+  let firstParentBox = wrapper.find(".parent_box");
+  let allParentBoxes = wrapper.findAll(".parent_box");
+  let sidebarItems = store.getters.sidebarItems;
+
+  let { path } = wrapper.vm.$router.currentRoute;
+
+  test("sidebarItemsLength to be equal to the length of all .parent_box", () => {
+    expect(
+      allParentBoxes.length
+    )
+    .toBe(
+      sidebarItems.length
+    );
+  });
+
+  test("correct icon renders", () => {
+    // checking for first item icon
+    expect(
+      sidebarItems[0].icon.split(" ")[1]
+    )
+    .toBe(
+      firstParentBox.find("i").classes()[1]
+    );
+  });
+
+  test("active slug corresponds to active sidebar item", () => {
+    expect(
+      wrapper.get(`.parent_box router-link-stub`).attributes('to')
+    )
+    .toBe(
+      path
+    );
+
+  });
+
+  test("sidebar toggles on click", async () => {
+    let beforeClick = wrapper.vm.$data.toggled;
```

Comment:
```suggestion
    const beforeClick = wrapper.vm.$data.toggled;
```
If you are going to change a variable directly later in a code block, use `let`. Else, use `const`.

In Javascript, as far as I know, it let you mutate an `const` declared variable/object ***properties***, but not the variable/object itself.

If you want to make both the variable/object and the properties immutable, see - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze